### DISCUSSION TITLE
How to sync window size between local & unix domain (attached first time)

### DISCUSSION THREAD

**Discussion by tranbinh1604** [wezterm/wezterm#6016] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hello, 

Im quite new with Wezterm but love it, atm im trying to sync the window size between the local domain and the unix domain when attached first time, this is what im getting with some configuration:

```
# wezterm.lua
config.unix_domains = { { name = "unix" } }
```

and use the key binding to attach

```
	{
		key = "a",
		mods = "LEADER",
		action = act.AttachDomain("unix"),
	},
```

Is there anyway to make the window size in unix domain alway fulfill all blank space in the screenshot? 

![CleanShot 2024-08-23 at 10 34 36@2x](https://github.com/user-attachments/assets/07cc43b8-1dfb-4a3a-a751-d56d980c6a54)

![CleanShot 2024-08-23 at 10 33 12](https://github.com/user-attachments/assets/612bd2bd-5d19-4ba0-ad1b-373f537d4a3f)

**Comment by Charlie284** [comment]
That gap is the imported tab keeping a smaller terminal grid; it isn't `window_padding`.

In `20240203-110809-5046fc22`, `AttachDomain` attaches first and, when the domain has no panes, [spawns its first tab using `initial_cols`/`initial_rows`](https://github.com/wezterm/wezterm/blob/5046fc225992db6ba2ef8812743fadfdfe4b184a/wezterm-gui/src/termwindow/mod.rs#L2921-L2943). The later reconciliation [takes the maximum of the GUI and imported tab sizes](https://github.com/wezterm/wezterm/blob/5046fc225992db6ba2ef8812743fadfdfe4b184a/wezterm-gui/src/termwindow/mod.rs#L1190-L1220), but it doesn't enlarge a smaller imported tab when that tab already has a nonzero DPI. [Current `main` still has that behavior](https://github.com/wezterm/wezterm/blob/b16aa977e32cfefb90a006e8b8a355ae4e5f1888/wezterm-gui/src/termwindow/mod.rs#L1257-L1285), so there isn't a config option that makes `AttachDomain` always stretch it.

For the first/empty-domain case, spawn directly into the named domain instead:

```lua
{
  key = 'a',
  mods = 'LEADER',
  action = act.SpawnCommandInNewTab {
    domain = { DomainName = 'unix' },
  },
},
```

That path implicitly attaches the detached domain and creates the new Unix-domain tab using the current tab's size, rather than `initial_cols`/`initial_rows` ([source](https://github.com/wezterm/wezterm/blob/5046fc225992db6ba2ef8812743fadfdfe4b184a/mux/src/lib.rs#L1282-L1325)). It is the cleanest option when the Unix mux is empty.

If the domain already contains persisted smaller tabs and you need to import those exact tabs, a real window resize (or fullscreen toggle) will reconcile the grid; this is the same workaround reported for the earlier mux sizing bug in [#2351](https://github.com/wezterm/wezterm/issues/2351#issuecomment-1364506994). There is no Lua `MuxTab:set_size()` method, so automating that case requires a delayed GUI-size nudge after `AttachDomain`, rather than another domain setting.
